cms/djangoapps/contentstore/tests/test_course_listing.py

```diff
@@ -107,6 +109,86 @@ def test_errored_course_global_staff(self):
             self.assertEqual(courses_list_by_groups, [])
 
+    @ddt.data(
+        (ModuleStoreEnum.Type.split, 5),
+        (ModuleStoreEnum.Type.mongo, 3)
+    )
+    @ddt.unpack
+    def test_staff_course_listing_performance(self, default_store, mongo_calls):
+        """
+        Create courses and compare the time to fetch accessible courses for the staff user and
+        normal user. When user is staff, fetched courses contains lightweight course summary objects and
+        contains CourseDescriptor objects otherwise. Fetching courses as a staff should take less time than
+        non-staff.
+        """
+
+        # Add user to the staff & create courses
+        GlobalStaff().add_users(self.user)
+        for num in xrange(TOTAL_COURSES_COUNT):
+            self._create_course_with_access_groups(
+                course_location=CourseLocator('Org', 'CreatedCourse' + str(num), 'Run'),
+                user=self.user,
+                store=default_store
+            )
+
+        # Fetch courses for staff user
+        courses_list_for_staff, __ = get_courses_accessible_to_user(self.request)
+
+        # Remove staff role of the user & measure courses access time.
+        GlobalStaff().remove_users(self.user)
+
+        # Fetch courses for non-staff user
+        courses_list_by_user, __ = get_courses_accessible_to_user(self.request)
+
+        # Verify returned courses as a staff & as non-staff are equal
+        self.assertEqual(len(courses_list_for_staff), len(courses_list_by_user), 'Returned Courses should be equal')
+
+        # Now count the db queries for staff
+        with check_mongo_calls(mongo_calls):
+            _staff_accessible_course_list(self.request)
+
+    @ddt.data(ModuleStoreEnum.Type.split, ModuleStoreEnum.Type.mongo)
+    def test_staff_course_listing(self, default_store):
+        """
+        Test that the fetch accessible courses list for staff and non-staff when get_course returns
+        CourseSummary instances for staff and CourseDescriptor for non-staff users.
+        """
+
+        # Assign & verify staff role to the user
+        GlobalStaff().add_users(self.user)
+        self.assertTrue(GlobalStaff().has_user(self.user))
+
+        # Create few courses
+        for num in xrange(USER_COURSES_COUNT):
+            self._create_course_with_access_groups(
+                course_location=CourseLocator('Org', 'CreatedCourse' + str(num), 'Run'),
+                user=self.user,
+                store=default_store
+            )
+
+        # Fetch accessible courses list & verify their count
+        courses_list_by_staff, __ = get_courses_accessible_to_user(self.request)
+        self.assertEqual(len(courses_list_by_staff), USER_COURSES_COUNT)
+
+        # Verify fetched accessible courses list is a list of CourseSummery instances
+        self.assertTrue(
+            [isinstance(course, CourseSummary) for course in courses_list_by_staff],
```

Comment:
You want:

```python
all(isinstance(course, CourseSummary) for course in courses_list_by_staff)
```
a list with something in it will always be truthy:

```python
>>> bool([False])
True
```